tests/devices/test_default_qubit.py

```diff
@@ -2527,3 +2528,76 @@ def test_error_hamiltonian_expval_finite_shots(self):
 
         with pytest.raises(AssertionError, match="Hamiltonian must be used with shots=None"):
             dev.expval(H)
+
+
+class TestBroadcastingSupport:
+    """Tests that the device correctly makes use of ``unbroadcast_expand`` to
+    execute broadcasted tapes."""
+
+    class RX_batched(qml.RX):
+        """A version of qml.RX that detects batching."""
+
+        ndim_params = (0,)
+        compute_decomposition = staticmethod(lambda theta, wires=None: qml.RX(theta, wires=wires))
+
+    @pytest.mark.parametrize("x", [0.2, [0.1, 0.6, 0.3], [0.1]])
+    @pytest.mark.parametrize("shots", [None, 100000])
+    def test_with_single_batched_par(self, x, shots):
+        """Test that broadcasting on a circuit with a
+        single parametrized operation works."""
+        dev = qml.device("default.qubit", wires=2, shots=shots)
+
+        @qml.qnode(dev)
+        def circuit(x):
+            self.RX_batched(x, wires=0)
+            return qml.expval(qml.PauliZ(0))
+            # return qml.expval(qml.Hamiltonian([0.3], [qml.PauliZ(0)]))
+
+        out = circuit(np.array(x))
+
+        assert circuit.device.num_executions == (1 if isinstance(x, float) else len(x))
+        tol = 1e-10 if shots is None else 1e-2
+        assert qml.math.allclose(out, qml.math.cos(x), atol=tol, rtol=0)
+
+    @pytest.mark.parametrize("x, y", [(0.2, [0.4]), ([0.1, 5.1], [0.1, -0.3])])
+    @pytest.mark.parametrize("shots", [None, 1000000])
+    def test_with_multiple_pars(self, x, y, shots):
+        """Test that broadcasting on a circuit with a
+        single parametrized operation works."""
+        dev = qml.device("default.qubit", wires=2, shots=shots)
+
+        @qml.qnode(dev)
+        def circuit(x, y):
+            self.RX_batched(x, wires=0)
+            self.RX_batched(y, wires=1)
+            return [qml.expval(qml.PauliZ(0)), qml.expval(qml.PauliY(1))]
+
+        out = circuit(x, y)
+        expected = qml.math.stack([qml.math.cos(x) * qml.math.ones_like(y), -qml.math.sin(y)]).T
+
+        assert circuit.device.num_executions == len(y)
+        tol = 1e-10 if shots is None else 1e-2
+        assert qml.math.allclose(out, expected, atol=tol, rtol=0)
+
+    @pytest.mark.parametrize("x, y", [(0.2, [0.4]), ([0.1, 5.1], [0.1, -0.3])])
+    @pytest.mark.parametrize("shots", [None, 1000000])
+    def test_with_Hamiltonian(self, x, y, shots):
+        """Test that broadcasting on a circuit with a
+        single parametrized operation works."""
+        dev = qml.device("default.qubit", wires=2, shots=shots)
+
+        H = qml.Hamiltonian([0.3, 0.9], [qml.PauliZ(0), qml.PauliY(1)])
+        H.compute_grouping()
+
+        @qml.qnode(dev)
+        def circuit(x, y):
+            self.RX_batched(x, wires=0)
+            self.RX_batched(y, wires=1)
+            return qml.expval(H)
```

Comment:
is it possible to batch `H`?

Reply:
E.g., 

```python
H = qml.Hamiltonian([np.array([0.1, 0.2]), np.array([1., -4.]), [qml.PauliZ(0), qml.PauliY(1)])
```

Reply:
Not yet... I think! :D This will depend also on the next PRs that allow to actually access numerical representations of operators and execute circuits on devices...
I'd hope to make this possible but have to check with all the processing of Hamiltonians...

Reply:
If I recall, this was a recent request of @cvjjm, hence why it was on my mind 😆